Python/fileutils.c

```diff
@@ -1274,7 +1274,12 @@ _Py_open_impl(const char *pathname, int flags, int gil_held)
 #endif
 
     if (gil_held) {
-        if (PySys_Audit("open", "sOi", pathname, Py_None, flags) < 0) {
+        PyObject *pathname_obj = PyUnicode_DecodeFSDefault(pathname);
```

Comment:
Would it be possible to call should_audit() here to avoid decoding the string if audit is disabled?

Something like:

```
if (should_audit()) {
  pathname_obj = PyUnicode_DecodeFSDefault(pathname);
  PySys_Audit(pathname_obj)
  Py_DECREF(pathname_obj);
}
```

Same remark for _Py_fopen().

Reply:
This is not performance critical, and `should_audit()` is not available in `Python/fileutils.c`, so this needs additional work. I prefer to merge simple fix. You can add the call `should_audit()` in separate PR.